File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/data/items/effects/ObjectEffectDuration.scala





















// Generated on 05/03/2013 10:35:06
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait ObjectEffectDuration extends ObjectEffect {
  override val typeId = 75

  protected trait Type extends ObjectEffect.Type with Serializable {
    override val typeId = ObjectEffectDuration.typeId

    val days: Short
    val hours: Short
    val minutes: Short

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeShort(days)
      buf.writeShort(hours)
      buf.writeShort(minutes)
    }

  }

  protected def create(actionId: Short, days: Short, hours: Short, minutes: Short): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val actionId = buf.readShort()
    val days = buf.readShort()
    val hours = buf.readShort()
    val minutes = buf.readShort()

    create(actionId, days, hours, minutes)
  }

}

object ObjectEffectDuration extends ObjectEffectDuration {
  case class TypeImpl(actionId: Short, days: Short, hours: Short, minutes: Short) extends Type
  override def create(actionId: Short, days: Short, hours: Short, minutes: Short) = TypeImpl(actionId, days, hours, minutes)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(actionId: Short, days: Short, hours: Short, minutes: Short) = create(actionId, days, hours, minutes)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/ObjectEffectDuration.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ObjectEffectDuration extends ObjectEffect {
    public static final short TYPE_ID = 75;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public short days;
    public short hours;
    public short minutes;
    
    public ObjectEffectDuration() { }
    
    public ObjectEffectDuration(short actionId, short days, short hours, short minutes) {
        super(actionId);
        this.days = days;
        this.hours = hours;
        this.minutes = minutes;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeShort(days);
        buf.writeShort(hours);
        buf.writeShort(minutes);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        days = buf.readShort();
        if (days < 0)
            throw new RuntimeException("Forbidden value on days = " + days + ", it doesn't respect the following condition : days < 0");
        hours = buf.readShort();
        if (hours < 0)
            throw new RuntimeException("Forbidden value on hours = " + hours + ", it doesn't respect the following condition : hours < 0");
        minutes = buf.readShort();
        if (minutes < 0)
            throw new RuntimeException("Forbidden value on minutes = " + minutes + ", it doesn't respect the following condition : minutes < 0");
    }
    
}
